src/llscan.cc

```diff
@@ -668,6 +672,59 @@ char** FindReferencesCmd::ParseScanOptions(char** cmd, ScanType* type) {
   return &cmd[optind - 1];
 }
 
+// Walk all contexts previously stored and print search_value_
+// reference if it exists. Not all values are associated with
+// a context object. It seems that Function-Local variables are
+// stored in the stack, and when some nested closure references
+// it is allocated in a Context object.
+void FindReferencesCmd::ReferenceScanner::PrintContextRefs(
+    SBCommandReturnObject& result, Error& err) {
+  ContextVector* contexts = llscan_->GetContexts();
+  v8::LLV8* v8 = llscan_->v8();
+
+  ContextVector visited_contexts;
+  for (auto ctx : *contexts) {
+    if (std::find(visited_contexts.begin(), visited_contexts.end(), ctx) !=
+        visited_contexts.end())
+      continue;
+    else
+      visited_contexts.push_back(ctx);
+
+    Error err;
+    v8::HeapObject context_obj(v8, ctx);
+    v8::Context c(context_obj);
+
+    v8::HeapObject scope_obj = c.GetScopeInfo(err);
+    if (err.Fail()) return;
+
+    v8::ScopeInfo scope(scope_obj);
+    v8::Smi param_count_smi = scope.ParameterCount(err);
+    if (err.Fail()) return;
+    v8::Smi stack_count_smi = scope.StackLocalCount(err);
+    if (err.Fail()) return;
+    v8::Smi local_count_smi = scope.ContextLocalCount(err);
+    if (err.Fail()) return;
+
+    int param_count = param_count_smi.GetValue();
+    int stack_count = stack_count_smi.GetValue();
+    int local_count = local_count_smi.GetValue();
+    for (int i = 0; i < local_count; i++) {
```

Comment:
I think we should move this logic to an iterator, since this is basically the same thing we have in https://github.com/nodejs/llnode/blob/master/src/llv8.cc#L1091-L1158.